### ISSUE TITLE
Include interaction data related to push-pull mechanism

### ISSUE THREAD

**Issue by mdrishti** [globalbioticinteractions/globalbioticinteractions#993]
Hi @jhpoelen ,
 
I am working on connecting data from globi with metabolite data from lotus & enpkg. I am using the 'interactions.tsv.gz' file from the globi website (https://www.globalbioticinteractions.org/data). I was testing some use-cases where such an integration could be useful for agriculture and push-pull mechanism stood out amongst others. But, when I try to find interactions specific to push-pull mechanism, I was able to find some like parasitoidOf, parasiteOf and hasHost, but not amongst the species I was expecting like _Desmodium_ and _Striga_. For instance, this paper has information on the specific interaction between these two species (https://www.sciencedirect.com/science/article/pii/S0031942215300376). There are many papers reporting the same observation like: 

https://www.sciencedirect.com/science/article/pii/S0031942215300376
https://www.sciencedirect.com/science/article/pii/S0261219406000329
https://scijournals.onlinelibrary.wiley.com/doi/full/10.1002/ps.6261
https://www.sciencedirect.com/science/article/pii/S0261219417300844?via%3Dihub

Maybe, there is some obvious interaction name that I am missing here. I was looking through the RO properties for something like "inhibits" or something in diseases relationships, but didn't find anything suitable to describe this interaction.

I think it would be great if globi could include such interactions. I could also help in curating information on this, but I am not really sure how to do that. I would appreciate any help in this direction.

Regards,
Disha

**Comment by jhpoelen** [comment]
Hi Disha @mdrishti - 

Thanks for your message and the references. Very neat to see this wealth of research done on how some plants inhibit the growth of others, and I hope that we can figure out a way to let GloBI index this valuable information.

And . . . it would be great if you could help curate this data. Perhaps you can follow the example by Stephen Fowler in https://www.globalbioticinteractions.org/2020/08/25/extracting-parasite-interaction-data/ . If their structure suits your needs to transcribe the relationships, then I'd suggest to copy their example and make transcribe some interactions from existing papers. I'd be happy to review and make sure that the data gets indexed by GloBI.

If you'd like to use other mechanisms to curate the interaction data extracted from the papers you mention, I'd suggest to have a look at https://globalbioticinteractions.org/contribute and https://github.com/globalbioticinteractions/template-dataset .  In my mind, whenever the data is (semi) structured, digital and openly accessible, GloBI should be able to index it.

@mdrishti Please let me know how you'd like to proceed.

@diatomsRcool @cmungall @cooperl09
 In your work on (plant) disease ontologies in specific and relation ontology in general, have you come across a good way to describe observations like:

```bibtex
 @article{Hooper_2015, title={Isolation and identification of Desmodium root exudates from drought tolerant species used as intercrops against Striga hermonthica}, volume={117}, ISSN={0031-9422}, url={http://dx.doi.org/10.1016/j.phytochem.2015.06.026}, DOI={10.1016/j.phytochem.2015.06.026}, journal={Phytochemistry}, publisher={Elsevier BV}, author={Hooper, A.M. and Caulfield, J.C. and Hao, B. and Pickett, J.A. and Midega, C.A.O. and Khan, Z.R.}, year={2015}, month=sep, pages={380–387} }
```

```[...] Plants from the genus Desmodium, in particular D. uncinatum, are used on sub-Saharan small-holder farms as intercrops to inhibit parasitism of cereal crops by Striga hermonthica and Striga asiatica via an allelopathic mechanism. [...]```

**Comment by jhpoelen** [comment]
@mdrishti looking forward to indexing your curated dataset using the push-pull mechanism.

You might want to consider using the term

http://purl.obolibrary.org/obo/RO_0002555 (allelopath of)

Would that fit your use?

**Comment by mdrishti** [comment]
Hi @jhpoelen,

I created the dataset here : https://github.com/mdrishti/pushPullIntxn. Please have a look and let me know if this is something GloBI can index.

Regards,
Disha

**Comment by jhpoelen** [comment]
@mdrishti thanks for your message. 

There's a bit of a delay between updates in github and the publicly available GloBI search index. 

In today's search results, I can see in the citation information of records sources from your datasets that the GloBI is using the late July 2024 version of your dataset (see below and attached screenshot. 

Thanks for being patient!

[📄](https://doi.org/10.1016/j.cropro.2006.01.008) [🔍](https://www.globalbioticinteractions.org/?interactionType=ecologicallyRelatedTo&accordingTo=https%3A%2F%2Fdoi.org%2F10.1016%2Fj.cropro.2006.01.008) [Khan ZR, Pickett JA, Wadhams LJ, Hassanali A, Midega CAO (2006). Combined control of Striga hermonthica and stemborers by maize–Desmodium spp. Intercrops. Crop Protection, 25:989-995 ](https://doi.org/10.1016/j.cropro.2006.01.008)Provider: [⚙️](https://github.com/mdrishti/pushPullIntxn) [🔍](https://www.globalbioticinteractions.org/?interactionType=ecologicallyRelatedTo&accordingTo=globi%3Amdrishti%2FpushPullIntxn) Tandon D. 2024. Push-pull interactions manually extracted from literature. Accessed via <https://github.com/mdrishti/pushPullIntxn/archive/7f473743a9261cf2eb989e08d1928fe03a8d5a82.zip> at 2024-07-27T12:52:06.844Z.  [discuss...](https://github.com/mdrishti/pushPullIntxn/issues/new?body=Hi%21%0A%0AThanks%20for%20helping%20to%20make%20existing%20biotic%20interaction%20data%20easier%20to%20find%20and%20access%21%0A%0AI%20was%20just%20looking%20at%20your%20GloBI%20indexed%20record%20at%20https%3A%2F%2Fwww.globalbioticinteractions.org%2F%3FaccordingTo%3Dglobi%253Amdrishti%252FpushPullIntxn%26interactionType%3DinteractsWith%20and%20I%20was%20wondering%20about%20...%20%28please%20add%20your%20own%20text%29&title=your%20indexed%20records%20for%20Khan%20ZR%2C%20Pickett%20JA%2C%20Wadhams%20LJ%2C%20Hassanali%20A%2C%20Midega%20CAO%20%282006%29.%20Combined%20control%20of%20Striga%20hermonthica%20and%20stemborers%20by%20maize%E2%80%93Desmodium%20spp.%20Intercrops.%20Crop%20Protection%2C%2025%3A989-995)

![image](https://github.com/user-attachments/assets/a304be44-0ab9-48e4-9920-521b6d6e7d2d)

**Comment by jhpoelen** [comment]
@mdrishti I just had a closer look at your dataset at https://github.com/mdrishti/pushPullIntxn and noticed that the review report indicated that no interactions were found by the GloBI review bots. 

So, I had a look at your dataset, and it appeared that some recent edits to globi.json caused the file to be malformed. Please review https://github.com/mdrishti/pushPullIntxn/pull/3 with my suggested changes. Also, please create a new release once you verify that the review report is producing expected results. 

Thanks again for being patient.

**Comment by mdrishti** [comment]
@jhpoelen, thanks for the modifications. I made a couple of adjustments. The review report looks good as far as interaction records are concerned. I have two questions:

a) obsevedDateTime column accepts only ISO-8601 formats which does not incorporate multiple durations like "long rainy seasons (March-August) 2001-2004". I would suggest including a column named observedDateTimeString where once can mention such data in strings, if possible.

b) the template tsv sheet contains info on habitats, but the final review report does not show information on habitats.

Also, the more I explore GloBI, the more I realize its importance. Thank you for this extremely useful resource!

**Comment by jhpoelen** [comment]
@mdrishti thanks for sharing your notes and for your attention to detail.

> a) obsevedDateTime column accepts only ISO-8601 formats which does not incorporate multiple durations like "long rainy seasons (March-August) 2001-2004". I would suggest including a column named observedDateTimeString where once can mention such data in strings, if possible. Four pushPull intxns, this information is important to be documented.

If you haven't already done so, please do add the column to your data. In my mind, datasets like yours drive the development of how GloBI interprets them.

> b) the template tsv sheet contains info on habitats, but the final review report does not show information on habitats.

Thanks for pointing out that the final review report does not show habitat information. Could you please help by opening a separate issue on this. Ideally, the issue should include steps to find what you are seeing (e.g., I looked at attached file x that I got from location y). In addition, I'd appreciate what you'd like to see instead. 

Thanks for your encouragement and for sharing your detailed notes. In my mind, interactions like the ones we are having are the things make make the GloBI community tick.

Thanks again for your patience in dealing with these configuration issues and review notes.

**Comment by mdrishti** [comment]
@jhpoelen, Came

**Comment by jhpoelen** [comment]
Hey @mdrishti - yes, the inverse interactions are inferred by GloBI in an attempt to facilitate search. When searching for hostOf interactions, I'd suspect that interactions original described as hasHost would also be of interest.

You caught my "hack" in inferring the allelopathOf - RO doesn't appear to have included an inverse, so I had to think of another one that fits. And, your comments make me realize that "hasPathogen" might not be a suitable inverse. 

I can see how asking for including hasAllelopath in RO would be one way to go. And, I'd be curious to hear more of your thoughts on better ways to make allelopathic interactions easier to find. As a more long term solution is put in place, one thing to do in the short term would be to change the inverse to be something  more suitable. 

Thanks again for taking the time to review GloBI interpretation of your data and sharing your detailed comments. Apologies for not getting it right the first time . . . and thanks for being patient.

Curious to hear your thoughts on how to proceed,
-jorrit

**Comment by jhpoelen** [comment]
@mdrishti is bringing up a lot of great point related to allelopathy, so now I am wondering - @cmungall @anitacaron @wdduncan any chance we can shortcut the usual review RO review process and add a inverse for existing "shortcut" interaction allelopathOf http://purl.obolibrary.org/obo/RO_0002555 ?

**Comment by diatomsRcool** [comment]
I'm just now getting around to this and seeing the question for me above. I think the allelopath terms are appropriate. I don't think adding the inverse of an existing term will require the same level of review as long as all the boxes are checked, but a PR is still needed.

**Comment by jhpoelen** [comment]
> @jhpoelen adding hasAllelopath as a new term sounds interesting. It will keep the original connection intact in inverse relations. I agree with suggesting this term in RO and I can raise an issue in their github.

@mdrishti It'd be great if you could propose the hasAllelopath addition in the RO github issue tracker. Would you feel comfortable preparing a Pull Request (PR) to introduce hasAllelopath ? If not please holler.

**Comment by jhpoelen** [comment]
> I am still curious : If I add inverse relations explicitly in the original dataset, will it stop GloBI from inferring them? If not, can we put a condition where it infers only when such interactions do not exist in the original dataset? I ask this because this can allow me (or other dataset creators) to have control over inverse interactions. Although, a point that works against adding control for inverse intxn could be that most RO terms have inverse properties (as you indicated) and adding ad-hoc control could be confusing. But still allelopathy is a gray area where adding control is useful IMO.

Right now, GloBI search index infers all inverse interactions by default. So, if an inverse interaction is explicitly defined, the inverse of the inverse would also be inferred. Are you proposing to let dataset curators let GloBI indexers (and others) know that inverse interactions should not be (or are preferred not to be) inferred? If so, please open a separate issue with some example on how you'd like to configure this in the globi.json index configuration file. If not, please elaborate on what you'd like GloBI's behavior to be.

**Comment by mdrishti** [comment]
> > @jhpoelen adding hasAllelopath as a new term sounds interesting. It will keep the original connection intact in inverse relations. I agree with suggesting this term in RO and I can raise an issue in their github.
> 
> @mdrishti It'd be great if you could propose the hasAllelopath addition in the RO github issue tracker. Would you feel comfortable preparing a Pull Request (PR) to introduce hasAllelopath ? If not please holler.

@jhpoelen Yes, I proposed the addition. I will create a PR.

**Comment by jhpoelen** [comment]
> @jhpoelen Yes, I proposed the addition. I will create a PR.

Thanks! If it is not too much trouble, could you you please link the issue/PR to this issue so that we can make sure that we tie up loose ends.

**Comment by mdrishti** [comment]
@jhpoelen, I started creating the NTR:has allelopath according to the instructions at https://github.com/oborel/obo-relations/blob/master/CONTRIBUTING.md#how-to-write-a-new-term-request. 

Stupid question (cuz I am still new to ontologies) - I kind of reached an obstacle, when I realized that the RO-id for a new term is not automatically generated. For instance, mine got named - RO_0000001. To me this feels odd. There are so many ids above 0000001 atleast numerically. On the other hand, RO_0000001 does not exist in obo-rel. I want to avoid taking an existing id and not really sure if there is a specific logic behind naming these ids. I would appreciate any help in this context. 
Maybe, I should have asked this directly at obo-rel, but I thought if you had some experience creating new terms, then I can ask you first.

-Disha
